Adding expression level histogram

ZhenghaoZhu opened the pull request:
TODO: Write details for implementation

![image](https://github.com/user-attachments/assets/e9dff157-17f5-47c3-9f97-7983390766d9)

enemeth19 commented:
I think we should remove the stratification by mutation status in the histogram so that the sole purpose of this plot is filtering based on expression level of the gene - what do you think?

enemeth19 commented:
Also, can you add the ability to filter the patient cohort by expression level (i.e., by dragging the cursor to select an area (but, really, a certain x-axis range) of the histogram?

ZhenghaoZhu commented:
Added the requested changes 
![image](https://github.com/user-attachments/assets/86693a4f-1bec-4a27-827f-6665ca3cc8fd)

enemeth19 commented:
This is a pretty plot that functions smoothly. Thanks so much for your work on this, Kevin. A few comments for ways that I think we can improve how it integrates into the rest of the app are below.
- This plot is visually more complex with more components than the pie chart, but all the components are meaningful, so for now I think we should keep its UI components as is, except for what’s noted below
- Regarding range slider:
(1) I think that how it currently works with zooming in on a part of the plot is the default behavior. I think it’d be simpler if, when a user drags their cursor on the plot, **_instead_ of zooming in to that area, the bars in that selected area are changed to a different color** (e.g., yellow, similar to the pie charts’ behavior which sort of ‘highlights’ the selection that was made). 
(2) I think that if part of a bar is captured in the dragged selection, the **patients contained in that _whole_ bar should be included in the selection**/filtered down cohort, since the bars represent a binning of the data 
- I think we should disable the bottom gray rectangle for now. I like it, but I think it makes the plot a bit too bulky. Maybe just comment the code lines that produce it and we can revisit it later
- The layout of all the plots in the data explore section are in disarray with the addition of this expression level plot. If we remove the bottom gray rectangle, maybe this will be restored. If not, then we’ll need to address it, perhaps as a separate issue
- The **cohort filtering** intended to occur via selections in the expression level plot **doesn’t get passed down to the downstream plots** after the submit button. We need to correct this as well

Thank you again, and please let us know your thoughts on these comments! Also happy to discuss a strategy/plan of attack for these changes.

enemeth19 commented:
Also, please modify this PR to merge into development branch rather than release branch

enemeth19 committed:
Gene expression histogram UI changes (per comments in PR):
- Revert to same bar colors and sizing as clinical feature histograms for uniformity
- Restrict plot selections to only be in horizontal direction (not vertical)
- Disable range slider
- Disable filter status div

enemeth19 commented:
The above commit to the _kevzhzh_histogram_ branch resolves point (1) described above pertaining to range slider, as well as:

> I think we should disable the bottom gray rectangle for now

enemeth19 committed:
Merge remote-tracking branch 'origin/development' into kevzhzh_histogram

enemeth19 committed:
Pass selected range in expression histogram as a filter criterion for expression data visualized in downstream plots (after Submit button)

enemeth19 commented:
The above [commit](https://github.com/web4bio/webgen/pull/499/commits/c78daf793c4e1c4118ec2f79a19f0e5a048c894b) to the _kevzhzh_histogram_ branch resolves the following: 
> The cohort filtering intended to occur via selections in the expression level plot doesn’t get passed down to the downstream plots after the submit button. We need to correct this as well

enemeth19 committed:
Rename createPieCharts.js to createDataExplorePlots.js (since this file generates both pie charts and histograms)

enemeth19 committed:
Refactor

adit-anand reviewed (approved):
Looks good to merge
